pool/pool_aio_test.go

```diff
@@ -1,5 +1,3 @@
-//go:build aiotest
```

Comment:
does it mean that a dev _has_ to have a docker for developing SDK?

Reply:
Correct. The same method already exists in rest gate and works perfect

Reply:
> already exists in the rest gate and works perfectly

sure, i do not think that your code or docker client lib is bad, i just meant that i am not sure we need to force its requirement for everybody: if i want to make a contribution to a `go` lib and run tests locally, i would expect me to have `go` and... all? while CI on the GH side can run everything with all the dependencies we want

what @roman-khimov thinks about it?

Reply:
Docker is a very common software, I tend to think almost all backend developers have it on board

Reply:
SDK is just a Go lib, i don't expect to require anything but pure Go to maintain it. If we wanna test AIO, lets test it within https://github.com/nspcc-dev/neofs-aio where Docker is natural

Reply:
Optional tests never really work, it has to be a part of the regular testing routine. I agree that in general it should be possible to cover the code with pure synthetic tests, but at the same time these are valuable as well, they prove that the system works as a whole.

What can be done is:
 * a separate target
 * but a mandatory GH action always running it

Maybe that's the optimal combination, but this still raises some questions of proper coverage counting.

Reply:
lets try to cover both goals. If we decide that Docker requirement can not be avoided, then i dont mind to have it if tests will really help

Reply:
I left `make test` as is and everyone may use it for SDK development. Also I've added a new make target `test-with-docker` for all tests, including docker AIO. Updated GH workflow to use `test-with-docker`